### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by bashamahmoud**

### CODE REVIEW

File: mysite/store/repos/cart_product_repo.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from ..models import CartProduct
+
+logger = logging.getLogger('store')
+
+
+class CartProductRepo:
+    @staticmethod
+    def get_cart_products(cart_pk=None, pk=None):
+        try:
+            if not pk:
```
Comment (MuhannadShamasneh) [inline mysite/store/repos/cart_product_repo.py:16]: If there is no pk, then do we still need to select related product?

File: mysite/store/repos/cart_product_repo.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                cart_products = CartProduct.objects.filter(cart_id=cart_pk).select_related('cart','product')
+            else:
+                cart_products = CartProduct.objects.filter(cart_id=cart_pk, product=pk).select_related('cart', 'product')
+            return cart_products
+        except Exception as e:
+            logger.error(f"Error retrieving cart: {e}")
+            return exception_handler(e, "Cart product")
+
+    @staticmethod
+    def create_cart_product(cart_product,product_id,cart_pk):
```
Comment (MuhannadShamasneh) [inline mysite/store/repos/cart_product_repo.py:26]: What if I need an api that create a new product and link it with the cart

File: mysite/store/repos/cart_product_repo.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            return cart_product.save(product_id=product_id,cart_id=cart_pk)
+        except Exception as e:
+            logger.error(f"Error creating cart product: {e}")
+            return exception_handler(e, "Cart product")
+
+    @staticmethod
+    def delete_cart_product(cart_pk=None, pk=None):
+        try:
+            cart_product = CartProduct.objects.filter(cart_id=cart_pk, product=pk).first()
+            return cart_product.delete()
```
Comment (MuhannadShamasneh) [inline mysite/store/repos/cart_product_repo.py:38]: Here you deleted the cart-product table, but what if the cart is empty, should we delete it?

File: mysite/store/serializers.py [hunk 4]
```diff
@@ -17,4 +17,20 @@ class Meta:
...
         model = Product
-        fields = '__all__'
+        fields = ['id', 'name', 'description', 'price', 'in_inventory']
+
+
+class CartSerializer(serializers.ModelSerializer):
+    products= ProductSerializer(many=True, read_only=True)
+    class Meta:
+        model = Cart
+        fields = ['id', 'customer', 'products']
```
Comment (MuhannadShamasneh) [inline mysite/store/serializers.py]: customer_id

File: mysite/store/serializers.py [hunk 5]
```diff
@@ -17,4 +17,20 @@ class Meta:
...
+
+
+class CartProductSerializer(serializers.ModelSerializer):
+    cart = CartSerializer(read_only=True)
+    product = ProductSerializer(read_only=True)
+    class Meta:
+        model = CartProduct
+        fields = ['id', 'quantity', 'cart', 'product']
```
Comment (MuhannadShamasneh) [inline mysite/store/serializers.py:35]: products

**Comment by bashamahmoud** [comment]
Fixed if statement error, stays on same functionality for create and delete

**Pull request merged**